client/virtualchannel_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client_test
+
+import (
+	"context"
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	channeltest "github.com/perun-network/perun-eth-backend/channel/test"
+	ethclienttest "github.com/perun-network/perun-eth-backend/client/test"
+	"perun.network/go-perun/channel"
+	ctest "perun.network/go-perun/client/test"
+	"polycry.pt/poly-go/test"
+)
+
+func TestVirtualChannelOptimistic(t *testing.T) {
+	rng := test.Prng(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	ctest.TestVirtualChannelOptimistic(
+		ctx,
+		t,
+		createVirtualChannelSetup(t, rng),
+	)
+}
+
+func TestVirtualChannelDispute(t *testing.T) {
+	rng := test.Prng(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	ctest.TestVirtualChannelDispute(
+		ctx,
+		t,
+		createVirtualChannelSetup(t, rng),
+	)
+}
+
+func createVirtualChannelSetup(t *testing.T, rng *rand.Rand) ctest.VirtualChannelSetup {
+	t.Helper()
+	clients, asset := setupVirtualChannelClients(t, rng)
```

Comment:
Couldn't you include the `setupVirtualChannelClients` code just inside this one or is it used somewhere else?

Reply:
I could, but what would be the benefit? Like this I think it is more readable.